### PR TITLE
Suppress incorrect warning in nargsort for timezone-aware DatetimeIndex

### LINKED ISSUE #25439: FutureWarning when sorting tz-aware datetimeindex
#### Problem
As reported [here](https://stackoverflow.com/questions/54854900/workaround-for-pandas-futurewarning-when-sorting-a-datetimeindex), sorting a tz-aware DatetimeIndex issues warning:

> FutureWarning: Converting timezone-aware DatetimeArray to timezone-naive ndarray with 'datetime64[ns]' dtype

MCVE here:
```python
import pandas as pd
idx = pd.DatetimeIndex(['2017-07-05 07:00:00', '2018-07-05 07:15:00','2017-07-05 07:30:00'])
df = pd.DataFrame({'C1':['a','b','c']},index=idx)
df = df.tz_localize('UTC')
df.sort_index()
```
#### Stack
The stack (Pandas 0.24.1) is:
```
__array__, datetimes.py:358
asanyarray, numeric.py:544
nargsort, sorting.py:257
sort_index, frame.py:4795
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.7.final.0
python-bits: 64
OS: Linux
OS-release: 4.18.9-acso
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_CA.UTF-8
LOCALE: en_CA.UTF-8

pandas: 0.24.1
pytest: None
pip: 19.0.2
setuptools: 40.7.2
Cython: 0.28.4
numpy: 1.16.1
scipy: None
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: None
patsy: None
dateutil: 2.8.0
pytz: 2018.9
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: 2.6.0
xlrd: 1.2.0
xlwt: None
xlsxwriter: None
lxml.etree: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
</details>

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #25439, alternative to https://github.com/pandas-dev/pandas/pull/25595 for 0.24.2

### CODE REVIEW

File: pandas/core/sorting.py [hunk 1]
```diff
@@ -254,7 +255,13 @@ def nargsort(items, kind='quicksort', ascending=True, na_position='last'):
             sorted_idx = np.roll(sorted_idx, cnt_null)
         return sorted_idx
 
-    items = np.asanyarray(items)
+    with warnings.catch_warnings():
+        # https://github.com/pandas-dev/pandas/issues/25439
+        # can be removed once ExtensionArrays are properly handled by nargsort
+        warnings.filterwarnings(
```
Comment (jreback) [inline pandas/core/sorting.py]: why don't we call ``.to_numpy()`` here instead? (I think think *always* is an index), or is this a performance issue? (until we have real EA support)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/sorting.py (hunk 1 above)
Comment (jreback) [reply pandas/core/sorting.py]: I am not opposed to this as an interim solution, just want to understand

Comment (jorisvandenbossche) [reply pandas/core/sorting.py]: As far as I understand, `to_numpy` has the future behaviour (i.e. converting to object dtype), while here we want to have the current M8[ns]

Comment (jorisvandenbossche) [reply pandas/core/sorting.py]: But of course, with `to_numpy(dtype='datetime64[ns]')`, you can get that.

In that case, we have to do a check here for datetimetz data; something like:

```
if is_datetimetz_dtype(items):
    items = items.to_numpy(dtype='datetime64[ns]')
else:
    items = np.asanyarray(items)
```

It's also custom code for this case, but if the above is preferable, happy to change (personally don't have a strong favor for either of them).

Comment (jreback) [reply pandas/core/sorting.py]: yeah i think this needs to be sorted, but ok for now with your existing.

Comment (jreback) [reply pandas/core/sorting.py]: however, shall we leave this or a follow-on open to fix this/

Comment (jorisvandenbossche) [reply pandas/core/sorting.py]: I would like to expand PR https://github.com/pandas-dev/pandas/pull/25595 to include a proper fix (for 0.25.0), involving a clean-up of EA.argsort, which will be the follow-up on this. 
And apparently, I already opened an issue about the argsort API problem a few months ago: https://github.com/pandas-dev/pandas/issues/21801 (will update that issue with the new information)

Comment (jreback) [reply pandas/core/sorting.py]: ok cool.

**Review by jreback — APPROVED** [review]

**Comment by TomAugspurger** [comment]
I'll put up a backport PR for this.

**Pull request merged**